tests/cpgrid/addLgrsOnDistributedGrid_test.cpp

```diff
@@ -588,3 +597,222 @@ BOOST_AUTO_TEST_CASE(globalRefine2)
         BOOST_CHECK_THROW(grid.globalRefine(1), std::logic_error);
     }
 }
+
+/* Commented due to restriction of fully interior LGRs. Used for developping a new approach to assign global ids when the
+   LGRs are distributed in different processes. To completely remove the assumption of fully interior LGRs, communication
+   is needed. 
+BOOST_AUTO_TEST_CASE(distributed_lgr)
+{
+    // Only for testing assignment of new global ids for refined entities (cells and point belonging to
+    // refined level grids).
+
+    // Create a grid
+    Dune::CpGrid grid;
+    const std::array<double, 3> cell_sizes = {1.0, 1.0, 1.0};
+    const std::array<int, 3> grid_dim = {4,3,3};
+    grid.createCartesian(grid_dim, cell_sizes);
+
+    std::vector<int> parts(36);
+    std::vector<std::vector<int>> cells_per_rank = { {0,1,4,5,8,9,16,20,21},
+                                                     {12,13,17,24,25,28,29,32,33},
+                                                     {2,3,6,7,10,11,18,22,23},
+                                                     {14,15,19,26,27,30,31,34,35} };
+    for (int rank = 0; rank < 4; ++rank) {
+        for (const auto& elemIdx : cells_per_rank[rank]) {
+            parts[elemIdx] = rank;
+        }
+    }
+    if(grid.comm().size()>1)
+    {
+        grid.loadBalance(parts);
```

Comment:
Not sure: Do we need to check whether grid.comm().size() >= 4 or is that done in loadBalance?

Reply:
It seems that I should check it, it's not done in loadBalance.